src/array/cuda/rowwise_sampling.cu

```diff
@@ -7,6 +7,8 @@
 #include <dgl/random.h>
 #include <dgl/runtime/device_api.h>
 #include <curand_kernel.h>
+#include <thrust/binary_search.h>
+#include <thrust/execution_policy.h>
```

Comment:
As you pointed out the issue with thrust versions, I think we should make sure the include path for the `third_party/thrust` is different than that for the one in the cuda toolkit, to ensure we are in fact including the correct headers.

Reply:
Added a CUB/Thrust version check. I changed the CMakeLists.txt to use external CUB/Thrust by default for consistent performance between CUDA versions, especially for `DeviceRadixSort`, `DeviceSegmentedSort`, etc.